connector/src/main/java/org/geysermc/connector/network/translators/world/block/entity/SkullBlockEntityTranslator.java

```diff
@@ -1,50 +1,168 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.connector.network.translators.world.block.entity;
+
+import com.github.steveice10.mc.auth.data.GameProfile;
+import com.github.steveice10.opennbt.tag.builtin.CompoundTag;
+import com.github.steveice10.opennbt.tag.builtin.ListTag;
+import com.github.steveice10.opennbt.tag.builtin.StringTag;
+import com.nukkitx.math.vector.Vector3f;
+import com.nukkitx.math.vector.Vector3i;
+import com.nukkitx.nbt.NbtMapBuilder;
+import com.nukkitx.protocol.bedrock.data.entity.EntityFlag;
+import org.geysermc.connector.entity.player.SkullPlayerEntity;
+import org.geysermc.connector.network.session.GeyserSession;
+import org.geysermc.connector.network.translators.world.block.BlockStateValues;
+import org.geysermc.connector.skin.SkinProvider;
+import org.geysermc.connector.skin.SkullSkinManager;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+@BlockEntity(name = "Skull", regex = "skull")
+public class SkullBlockEntityTranslator extends BlockEntityTranslator implements RequiresBlockState {
+    public static boolean ALLOW_CUSTOM_SKULLS;
+
+    @Override
+    public boolean isBlock(int blockState) {
+        return BlockStateValues.getSkullVariant(blockState) != -1;
+    }
+
+    @Override
+    public void translateTag(NbtMapBuilder builder, CompoundTag tag, int blockState) {
+        byte skullVariant = BlockStateValues.getSkullVariant(blockState);
+        float rotation = BlockStateValues.getSkullRotation(blockState) * 22.5f;
+        // Just in case...
+        if (skullVariant == -1) {
+            skullVariant = 0;
+        }
+        builder.put("Rotation", rotation);
+        builder.put("SkullType", skullVariant);
+    }
+
+    public static GameProfile getProfile(CompoundTag tag) {
+        if (tag.contains("SkullOwner")) {
+            CompoundTag owner = tag.get("SkullOwner");
+            CompoundTag properties = owner.get("Properties");
+            if (properties == null) {
+                try {
+                    CompletableFuture<GameProfile> gameProfile = SkinProvider.requestTexturesFromUsername(owner);
+                    return gameProfile.get();
```

Comment:
Would recommend storing the result if you need it more then once, and maybe add a notice that the get method blocks the thread?
Personally I would make the getProfile method async and spawn the initial skull after the getProfile has been completed.